src/mantle/mantle_query_sync.c

```diff
@@ -75,3 +76,133 @@ GR_RESULT grWaitForFences(
         return GR_ERROR_OUT_OF_MEMORY;
     }
 }
+
+
+GR_RESULT grGetFenceStatus(
+    GR_FENCE fence)
+{
+    GrFence* grFence = (GrFence*)fence;
+    if (fence == NULL){
+	return GR_ERROR_INVALID_POINTER;
+    }
```

Comment:
add check for `grFence->sType == GR_STRUCT_TYPE_FENCE` that returns `GR_ERROR_INVALID_OBJECT_TYPE` on failure

Reply:
same in other functions.

Reply:
fixed, also fixed it in grWaitForFences